src/routes/docs/quick-starts/oauth-google/+page.markdoc

```diff
@@ -0,0 +1,353 @@
+
+## Enabling the Google provider {% #enable-provider %}
+
+Before you can use Google to authenticate users, you need to enable the provider in your Appwrite console.
+
+1. Navigate to your Appwrite project
+2. Navigate to **Auth** > **Settings**
+3. Find and open the OAuth provider
+4. In the **Google OAuth2 Settings** modal, use the toggle to enable the provider
+
+Don't close this modal, we'll need an OAuth client ID to complete this form.
+
+## Creating a Google OAuth2.0 client  {% #create-client %}
+
+To use Google OAuth with Appwrite, you need to create an OAuth 2.0 client from your Google Cloud console. 
+You can do this by following the [Setting up OAuth 2.0](https://support.google.com/googleapi/answer/6158849) guide from Google.
+
+1. When prompted to provide an **Application type**, select **web application**, irrespective of your target platform. This is because the communication occurs between Appwrite (a web app) and the OAuth2 provider on behalf of your web or mobile app.
+1. When prompted to provide a **Authorized redirect URIs** field, provide the **URI** found in the **Google OAuth2 Settings** modal from your Appwrite console.
+1. After you've created the OAuth client, you can head back to your Appwrite console to complete the form in the **Google OAuth2 Settings** modal.
+1. Find the **Client ID** in OAuth 2.0 Client IDs and provide this in the **App ID** field in the **Google OAuth2 Settings** modal from the Appwrite console.
+1. Navigate to the **Client secrets** section in the newly created client, copy the Client secret and provide this in the **App Secret** field in the **Google OAuth2 Settings** modal from the Appwrite console.
+
+## Authenticating {% #authenticating %}
+
+You can use any of the Appwrite Client SDKs to authenticate users with their Google account.
+
+{% tabs %}
+{% tabsitem #js title="Javascript" %}
+When a user calls the [Create OAuth2 Session](https://appwrite.io/docs/client/account#accountCreateOAuth2Session) endpoint in your web app, they will be taken to Google's OAuth page to complete their login.
+
+After authenticating, they'll be redirected back to your app using either the `success` or `failure` URLs provided. To provide the best experience to your users, make sure to **implement and provide both routes** to prompt the user about successful and failed authentication attempts.
+
+```js
+import { Client, Account } from "appwrite";
+
+const client = new Client();
+
+const account = new Account(client);
+
+client
+  .setEndpoint("https://cloud.appwrite.io/v1") // Your API Endpoint
+  .setProject("[PROJECT_ID]");                 // Your project ID
+
+// Go to OAuth provider login page
+account.createOAuth2Session("google", "[LINK_ON_SUCCESS]", "[LINK_ON_FAILURE]");
+```
+{% /tabsitem %}
```

Comment:
We should probably add a note about how they might need to configure a custom domain if 3rd party cookies are being blocked.